nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,411 @@
+            } else {
+                // Verify that the last connection was not dequeued but created as new, because it exceeded the queue size limit
+                assertThat(connection, is(not(connectionList.get(i))));
+            }
+        }
+
+        // The queue is currently empty so check if we can add the last created connection into it.
+        response.close();
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        ClientConnection connectionNow = ((ClientRequestImpl) request).getConnection(true);
+        request.connection(connectionNow);
+        Http1ClientResponse responseNow = request.request();
+        // Verify that the connection was dequeued
+        assertThat(connectionNow, is(connection));
+    }
+
+    private static void validateSuccessfulResponse(Http1Client client) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        Http1ClientResponse response = request.submit(requestEntity);
+
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(response.entity().as(String.class), is(requestEntity));
+    }
+
+    private static void validateFailedResponse(Http1Client client, String errorMessage) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        final IllegalStateException ie =
+                assertThrows(IllegalStateException.class, () -> request.submit(requestEntity));
+        assertThat(ie.getMessage().contains(errorMessage), is(true));
+    }
+
+    private void validateChunkTransfer(Http1ClientResponse response, boolean chunked, long contentLength, String entity) {
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(Long.parseLong(response.headers().get(REQ_CONTENT_LENGTH_HEADER_NAME).value()), is(contentLength));
+        if (chunked) {
+            assertThat(response.headers(), hasHeader(REQ_CHUNKED_HEADER));
+        } else {
+            assertThat(response.headers(), noHeader(REQ_CHUNKED_HEADER.headerName()));
+        }
+        String responseEntity = response.entity().as(String.class);
+        assertThat(responseEntity, is(entity));
+    }
+
+    private static void responseHandler(ServerRequest req, ServerResponse res) {
+        customHandler(req, res, false);
+    }
+
+    private static void chunkResponseHandler(ServerRequest req, ServerResponse res) {
+        customHandler(req, res, true);
+    }
+
+    private static void customHandler(ServerRequest req, ServerResponse res, boolean chunkResponse) {
+        Headers reqHeaders = req.headers();
+        if (reqHeaders.contains(Http.HeaderValues.EXPECT_100)) {
+            res.headers().add(REQ_EXPECT_100_HEADER_NAME);
```

Comment:
`add` will check if the header exists and then add a value to it if it does.
method `set` is a better fit in this case, as it replaces a value if any exists

Reply:
Done